include/seastar/util/attribute-compat.hh

```diff
@@ -21,7 +21,8 @@
 #pragma once
 
 #if defined(__has_cpp_attribute) && __has_cpp_attribute(nodiscard)
+    #warning "SEASTAR_NODISCARD is deprecated. Use [[nodiscard]]."
```

Comment:
What's the point of the warning? It will be shown whether or not someone uses SEASTAR_NODISCARD.

We can just remove attribute-compat.hh.

Reply:
okay. wanted to be more gentle to developers who are using this macro. but yeah, the inner me really wants to drop this file instead. will do.